includes/class-wc-payments.php

```diff
@@ -1291,7 +1291,15 @@ public static function register_checkout_gateway( $payment_method_registry ) {
 	 * Handles upgrade routines.
 	 */
 	public static function install_actions() {
-		if ( version_compare( WCPAY_VERSION_NUMBER, get_option( 'woocommerce_woocommerce_payments_version' ), '>' ) ) {
+		$dependency_service = self::$dependency_service;
+
+		// Make sure we have correct class.
+		if ( ! $dependency_service instanceof WC_Payments_Dependency_Service ) {
+			$dependency_service = new WC_Payments_Dependency_Service();
+		}
```

Comment:
I know we're just trying to be defensive here, but I don't think it's possible for this to be null here?

What do you think of the trade-off between the complexity here defending against something that might change in the future?

Reply:
Upon this point, I was scared and afraid of random nulls 😄  But you are correct.